### PR TITLE
Implement Orders API on tickets route

### LINKED ISSUE #660: Implement orders API on tickets route
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

image

### PR DESCRIPTION
**Pull request by CosmicCoder96**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Implements Orders API on tickets route

Fixes #660

### CODE REVIEW

**Comment by niranjan94** [comment]
Deployment link: https://pr-734-open-event-frontend-fossasia.surge.sh

File: app/templates/events/view/tickets/orders.hbs [hunk 1]
```diff
@@ -51,7 +51,10 @@
   <div class="row">
     <div class="eight wide column">
       {{#tabbed-navigation isNonPointing=true}}
-        {{#link-to 'events.view.tickets.orders.index' class='item'}}
+        {{#link-to 'events.view.tickets.orders.list' 'all' class='item'}}
+          {{t 'All'}}
+        {{/link-to}}
+        {{#link-to 'events.view.tickets.orders.list' 'all' class='item'}}
```
Comment (geekyd) [inline app/templates/events/view/tickets/orders.hbs]: @CosmicCoder96 This should be `completed` right?

File: app/routes/events/view/tickets/orders/list.js [hunk 2]
```diff
@@ -20,30 +20,8 @@ export default Route.extend({
...
-        'id'            : 2,
-        'quantity'      : 1,
-        'invoiceNumber' : '#O1345883292-302',
-        'amount'        : 50,
-        'paidVia'       : 'Paypal',
-        'buyer'         : 'sample@gmail.com',
-        'buyerName'     : 'Sample Buyer2'
-      }
-    ];
+    return this.modelFor('events.view').query('orders', {
```
Comment (geekyd) [inline app/routes/events/view/tickets/orders/list.js]: @CosmicCoder96 Please use filter.

File: app/controllers/events/view/tickets/orders/list.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      title          : 'Order',
+      disableSorting : true
+    },
+    {
+      propertyName : 'amount',
+      template     : 'components/ui-table/cell/events/view/tickets/orders/cell-amount',
+      title        : 'Total Amount'
+    },
+    {
+      propertyName : 'tickets',
```
Comment (geekyd) [inline app/controllers/events/view/tickets/orders/list.js]: Make this tickets.length & remove the cell-quantity template.

**Comment by niranjan94** [comment]
Deployment link: https://pr-734-open-event-frontend-fossasia.surge.sh

**Comment by CosmicCoder96** [comment]
@geekyd Done !

File: app/controllers/events/view/tickets/orders/list.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      propertyName : 'amount',
+      template     : 'components/ui-table/cell/events/view/tickets/orders/cell-amount',
+      title        : 'Total Amount'
+    },
+    {
+      propertyName : 'tickets.length',
+      title        : 'Quantity'
+    },
+    {
+      propertyName   : 'user',
```
Comment (geekyd) [inline app/controllers/events/view/tickets/orders/list.js]: @CosmicCoder96 We can directly pass the email here `user.email`, we do not need an extra template for it.

File: app/templates/components/ui-table/cell/events/view/tickets/orders/cell-amount.hbs [hunk 5]
```diff
@@ -0,0 +1 @@
+US$ {{number-format record.amount}}
```
Comment (geekyd) [inline app/templates/components/ui-table/cell/events/view/tickets/orders/cell-amount.hbs]: Reuse the template `cell/admin/sales/status/cell-amount`

File: app/routes/events/view/tickets/orders/list.js [hunk 6]
```diff
@@ -20,30 +20,54 @@ export default Route.extend({
...
-        'quantity'      : 1,
-        'invoiceNumber' : '#O1345883292-302',
-        'amount'        : 50,
-        'paidVia'       : 'Paypal',
-        'buyer'         : 'sample@gmail.com',
-        'buyerName'     : 'Sample Buyer2'
-      }
-    ];
+    let filterOptions = [];
+    if (params.orders_status === 'completed') {
```
Comment (geekyd) [inline app/routes/events/view/tickets/orders/list.js]: We can replace it by 
```
return this.modelFor('events.view').query('orders', {
  include      : 'tickets,user',
  filter           :[{
   name : 'status',
   op      : 'eq',
   val     : params.orders_status
}] ,
  'page[size]' : 10
 });
```

Right?

File: app/templates/components/ui-table/cell/events/view/tickets/orders/cell-amount.hbs (hunk 5 above)
Comment (CosmicCoder96) [inline app/templates/components/ui-table/cell/events/view/tickets/orders/cell-amount.hbs]: @geekyd Won't that seem counter intuitive ? To use an admin/sales/ template, maybe we can let this be and later in a refactor take care of the amount fields used  everywhere

File: app/routes/events/view/tickets/orders/list.js (hunk 6 above)
Comment (CosmicCoder96) [inline app/routes/events/view/tickets/orders/list.js]: @geekyd What about the case when the `all` tab is selected?

File: app/templates/components/ui-table/cell/events/view/tickets/orders/cell-amount.hbs (hunk 5 above)
Comment (geekyd) [inline app/templates/components/ui-table/cell/events/view/tickets/orders/cell-amount.hbs]: Okay :)

File: app/routes/events/view/tickets/orders/list.js (hunk 6 above)
Comment (geekyd) [inline app/routes/events/view/tickets/orders/list.js]: Yeah, we can split it into two cases then. What do you think?

Comment (CosmicCoder96) [inline app/routes/events/view/tickets/orders/list.js]: already done :)

**Comment by niranjan94** [comment]
Deployment link: https://pr-734-open-event-frontend-fossasia.surge.sh

**Pull request merged**